### PR TITLE
Make town and nation level related errors easier to debug

### PR DESCRIPTION
**Pull request by Warriorrrr**
#### Description: 

____
#### New Nodes/Commands/ConfigOptions: 

____
#### Relevant Towny Issue ticket:

____
- [ ] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

File: src/com/palmergames/bukkit/towny/TownySettings.java [hunk 1]
```diff
@@ -207,25 +211,25 @@ public static void loadTownLevelConfig() throws IOException {
...
-					Integer.parseInt(level.get("townOutpostLimit").toString()),
-					Integer.parseInt(level.get("townBlockBuyBonusLimit").toString()),
-					Double.parseDouble(level.get("debtCapModifier").toString()),
-					Double.parseDouble(level.get("resourceProductionModifier").toString()),
-					Double.parseDouble(level.get("bankCapModifier").toString()),
+					levelGetAndParse(level, "town levels", numResidentsIndex, "numResidents", null, Integer::parseInt),
+					levelGet(level, "nation levels", numResidentsIndex, "namePrefix", ""),
+					levelGet(level, "nation levels", numResidentsIndex, "namePostfix", ""),
+					levelGet(level, "nation levels", numResidentsIndex, "mayorPrefix", ""),
+					levelGet(level, "nation levels", numResidentsIndex, "mayorPostfix", ""),
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/TownySettings.java:218]: these "nation levels" should be "town levels" for the correct error messaging.

**Review by LlmDl — CHANGES_REQUESTED** [review]
I would also be nice to know that the loading of town/nation levels still works with the config migrator, now that it isn't all using strings.

**Comment by Warriorrrr** [comment]
The loading is still essentially the same so I assume the config migrator will play nice, levelGet is equivalent to level.get("key").toString() but with some nice error messaging

**Comment by LlmDl** [comment]
Looks good to me:
```
[16:25:28] [Server thread/INFO]: [Towny] Config: 0.98.6.3 applying 1 automatic update...
[16:25:28] [Server thread/INFO]: [Towny] Config: 0.98.6.5 applying 2 automatic updates...
[16:25:28] [Server thread/INFO]: [Towny] Config: 0.98.6.7 applying 3 automatic updates...
```

**Pull request merged**